### ISSUE TITLE
Version incompatibility with latest scikit-learn.

### ISSUE THREAD

**Issue by ytsmiling** [scikit-optimize/scikit-optimize#902]
```python -c "import skopt"``` fails when `scikit-learn==0.23.0` is installed.

## Error messages
```
ImportError: cannot import name 'MaskedArray' from 'sklearn.utils.fixes' (~/.pyenv/versions/3.8.2/lib/python3.8/site-packages/sklearn/utils/fixes.py)
```

## Environment
I reproduced the problem at least in the following python versions
python: `3.6.6`, `3.8.2`

## Minimum code to reproduce
```bash
pip install scikit-learn==0.23.0 scikit-optimize==0.7.4
python -c "import skopt"
```

**Comment by Innixma** [comment]
Can confirm that this breaks `autogluon<=0.0.8` with `import autogluon`

**Comment by Innixma** [comment]
Issue on scikit-learn: https://github.com/scikit-learn/scikit-learn/issues/17213

**Comment by rth** [comment]
A temporary workaround for people experiencing this would be to monkeypatch the missing import,
```py
from numpy.ma import MaskedArray
import sklearn.utils.fixes

sklearn.utils.fixes.MaskedArray = MaskedArray

import skopt
```

A proper fix will be included in scikit-learn 0.23.1 (or if scikit-optimize makes a release with a fix from https://github.com/scikit-optimize/scikit-optimize/pull/904 before).

**Comment by magic-lantern** [comment]
I am using skopt 0.7.4 and scikit-learn 0.23.1 and looks like 0.23.1 introduced some new errors.

Here's some example code taken straight from the website at https://scikit-optimize.github.io/stable/auto_examples/sklearn-gridsearchcv-replacement.html#minimal-example:

```python
from skopt import BayesSearchCV
from sklearn.datasets import load_digits
from sklearn.svm import SVC
from sklearn.model_selection import train_test_split

X, y = load_digits(10, True)
X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.75, test_size=.25, random_state=0)

# log-uniform: understand as search over p = exp(x) by varying x
opt = BayesSearchCV(
    SVC(),
    {
        'C': (1e-6, 1e+6, 'log-uniform'),
        'gamma': (1e-6, 1e+1, 'log-uniform'),
        'degree': (1, 8),  # integer valued parameter
        'kernel': ['linear', 'poly', 'rbf'],  # categorical parameter
    },
    n_iter=32,
    cv=3
)

opt.fit(X_train, y_train)

print("val. score: %s" % opt.best_score_)
print("test score: %s" % opt.score(X_test, y_test))
```

When I run this example, I get the following output:

    ---------------------------------------------------------------------------
    TypeError                                 Traceback (most recent call last)
    in 
        20 )
        21 
    ---> 22 opt.fit(X_train, y_train)
        23 
        24 print("val. score: %s" % opt.best_score_)

    /opt/anaconda3/envs/py37/lib/python3.7/site-packages/skopt/searchcv.py in fit(self, X, y, groups, callback)
        678                 optim_result = self._step(
        679                     X, y, search_space, optimizer,
    --> 680                     groups=groups, n_points=n_points_adjusted
        681                 )
        682                 n_iter -= n_points

    /opt/anaconda3/envs/py37/lib/python3.7/site-packages/skopt/searchcv.py in _step(self, X, y, search_space, optimizer, groups, n_points)
        564         refit = self.refit
        565         self.refit = False
    --> 566         self._fit(X, y, groups, params_dict)
        567         self.refit = refit
        568 

    /opt/anaconda3/envs/py37/lib/python3.7/site-packages/skopt/searchcv.py in _fit(self, X, y, groups, parameter_iterable)
        477                 # `"param_%s" % name` at the first occurence of `name`.
        478                 # Setting the value at an index also unmasks that index
    --> 479                 param_results["param_%s" % name][cand_i] = value
        480 
        481         results.update(param_results)

    /opt/anaconda3/envs/py37/lib/python3.7/site-packages/sklearn/utils/deprecation.py in wrapped(*args, **kwargs)
        66         def wrapped(*args, **kwargs):
        67             warnings.warn(msg, category=FutureWarning)
    ---> 68             return init(*args, **kwargs)
        69         cls.__init__ = wrapped
        70 

    TypeError: object.__init__() takes exactly one argument (the instance to initialize)

If I downgrade scikit-learn back to 0.23.0 and do the monkey patch as described above, everything works ok.

**Comment by holgern** [comment]
There are also changes in Gaussian processes regression in sklearn 0.23 which needs to be migrated:
https://github.com/scikit-learn/scikit-learn/pull/15782

**Comment by GuillemGSubies** [comment]
Any updates on this? Can I help in any way?

**Comment by xmatthias** [comment]
scikit-learn==0.23.1  does solve this for now.
they did properly deprecate the failing option now instead of simply removing: 
source: https://github.com/scikit-learn/scikit-learn/pull/17199

**Comment by GuillemGSubies** [comment]
But the error `TypeError: object.__init__() takes exactly one argument (the instance to initialize)` Is not solved with that, is it?

**Comment by xmatthias** [comment]
but that's not what this issue is about originally, that was just attached to this issue.

**Comment by magic-lantern** [comment]
I added the information about `TypeError: object.__init__() takes exactly one argument` as it is part of scikit-optimize not being compatible with the most recent scikit-learn. Additionally some comments above said that the newest scikit-learn should remove the need for monkey patching.  If it is preferred I can open a new issue (if the issue still exists).

**Comment by amueller** [comment]
I think this is actually part of the original issue as it was meant to be a fix, but I think there's an issue with the fix. If you want to look into it, I think we should fix it in sklearn.

**Comment by ZhaoJunLee** [comment]
Instead of 
from **skopt** import BayesSearchCV

try,
from **skopt.searchcv** import BayesSearchCV

this works well for me.

**Comment by epx-dan** [comment]
I am using scikit-survival for building CoxPH survival model and BayesSearchCV for tunning, however, I have error `TypeError: object.__init__() takes no parameters` when I try to run `BayesSearchCV(estimator=CoxPHSurvivalAnalysis(), search_spaces=params_coxph, cv=10, n_jobs=-1).fit(X_train, y_train)`. How to solve this?

**Comment by holgern** [comment]
Version incompatibility has been fixed in #943 and #939 and in release 0.8.1
